lib/Doctrine/DBAL/Driver/Mysqli/MysqliStatement.php

```diff
@@ -106,6 +113,8 @@ public function __construct(\mysqli $conn, $prepareString)
             $this->types = str_repeat('s', $paramCount);
             $this->_bindedValues = array_fill(1, $paramCount, null);
         }
+
+        $this->lastInsertId = $lastInsertId;
```

Comment:
That won't help as we need the reference in the connection. I just wanted to keep BC here, but not sure if that is necessary.